gateway/reverse_proxy_test.go

```diff
@@ -132,7 +132,7 @@ type configTestReverseProxyDnsCache struct {
 	dnsConfig   config.DnsCacheConfig
 }
 
-func (s *Test) SetupTestReverseProxyDnsCache(cfg *configTestReverseProxyDnsCache) func() {
+func (s *Test) flakySetupTestReverseProxyDnsCache(cfg *configTestReverseProxyDnsCache) func() {
```

Comment:
I renamed the function to clearly mark the function as being flaky, the tests failed within this function.